src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,71 @@
+package storage;
+
+import tasks.Task;
+import tasks.TaskList;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.ObjectInputStream;
+import java.io.FileOutputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Storage components which handles loading and saving of file.
+ */
+public class Storage {
+    // deals with loading tasks from the file and saving tasks in the file
+    private String filePath;
+
+    /**
+     * Constructor of a storage component.
+     * @param filePath Path of file we interact with.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns a list of tasks.
+     * @return TaskList which is an abstraction of our task list.
+     */
+    public TaskList load() {
+        TaskList tasks = new TaskList(new ArrayList<Task>(), true);
+        try {
+            File file = new File(this.filePath);
+            if (!file.exists()) {
+                File directory = new File(file.getParent());

```

Comment:
I love how you handled the situation where a file or its parent directory/directories may not exist! Very clean code!